sophia-vote-contract.md

```diff
@@ -61,19 +61,25 @@ We need to do this because in Sophia ML we don not have a default value of 0x0/f
 
 Next we create the vote function which looks basically like this:
 `````
-public stateful function vote(candidate: address) : bool =       is_candidate(candidate)       let new_votes_state = Call.caller :: state.votes[candidate].voters       put(state{          votes[candidate].voters = new_votes_state }) true
+stateful entrypoint vote(candidate: address) : bool =
+   is_candidate(candidate)
+   let new_votes_state = Call.caller :: state.votes[candidate].voters
+   put(state{
+      votes[candidate].voters = new_votes_state })
+   true
 ``````
 
 We access the transaction initiator’s address by the built in `Call.caller` and prepend it `::` to the current list of voters.
 
 The last step is to create a `get votes count` function.
 `````
-public function count_votes(candidate: address) : int =       size(state.votes[candidate].voters)
+entrypoint count_votes(candidate : address) : int =
+   size(state.votes[candidate].voters)  
 `````
 
 We need to make use of a `size` function which we define as a helper function below. Here is the code:
 `````
-private function size(l : list('a)) : int = size'(l, 0)
+function size(l : list('a)) : int = size'(l, 0)
```

Comment:
The size function is now included with the standard library

`include "List.aes"`

https://github.com/aeternity/protocol/blob/master/contracts/sophia_stdlib.md#length

Reply:
Thanks for the feedback. I tried a couple of syntaxes which returns `unbound variable...` error

**Tried syntaxes**
```
List.length(state.votes[candidate].voters)
length(state.votes[candidate].voters)
List.size(state.votes[candidate].voters)
size(state.votes[candidate].voters)
```

Kindly assist with this

Reply:
I  failed to `include "List.aes". It is working now.